src/core_codemods/https_connection.py

```diff
@@ -1,135 +1,82 @@
-        NAME="https-connection",
-        REVIEW_GUIDANCE=ReviewGuidance.MERGE_WITHOUT_REVIEW,
-        REFERENCES=[
-            {
-                "url": "https://owasp.org/www-community/vulnerabilities/Insecure_Transport",
-                "description": "",
-            },
-            {
-                "url": "https://urllib3.readthedocs.io/en/stable/reference/urllib3.connectionpool.html#urllib3.HTTPConnectionPool",
-                "description": "",
-            },
-        ],
-    )
-    CHANGE_DESCRIPTION = METADATA.DESCRIPTION
-    SUMMARY = (
-        "Changes HTTPConnectionPool to HTTPSConnectionPool to Enforce Secure Connection"
-    )
+
+class HTTPSConnectionModifier(ImportedCallModifier):
+    def updated_args(self, original_args):
+        """
+        Last argument _proxy_config does not match new method
+
+        We convert it to keyword
+        """
+        new_args = list(original_args)
+        if self.count_positional_args(new_args) == 10:
+            new_args[9] = new_args[9].with_changes(
+                keyword=cst.parse_expression("_proxy_config")
+            )
+        return new_args
+
+    def update_attribute(self, true_name, original_node, updated_node, new_args):
+        del true_name, original_node
+        return updated_node.with_changes(
+            args=new_args,
+            func=updated_node.func.with_changes(
+                attr=cst.parse_expression("HTTPSConnectionPool")
+            ),
+        )
+
+    def update_simple_name(self, true_name, original_node, updated_node, new_args):
+        del true_name
+        AddImportsVisitor.add_needed_import(self.context, "urllib3")
+        RemoveImportsVisitor.remove_unused_import_by_node(self.context, original_node)
+        return updated_node.with_changes(
+            args=new_args,
+            func=cst.parse_expression("urllib3.HTTPSConnectionPool"),
+        )
+
+    def count_positional_args(self, arglist: Sequence[cst.Arg]) -> int:
+        for i, arg in enumerate(arglist):
+            if arg.keyword:
+                return i
+        return len(arglist)
+
+
+class HTTPSConnection(BaseCodemod):
+    SUMMARY = "Enforce HTTPS Connection for `urllib3`"
+    NAME = "https-connection"
+    REVIEW_GUIDANCE = ReviewGuidance.MERGE_WITHOUT_REVIEW
+    REFERENCES = [
+        {
+            "url": "https://owasp.org/www-community/vulnerabilities/Insecure_Transport",
+            "description": "",
+        },
+        {
+            "url": "https://urllib3.readthedocs.io/en/stable/reference/urllib3.connectionpool.html#urllib3.HTTPConnectionPool",
+            "description": "",
+        },
+    ]
 
     METADATA_DEPENDENCIES = (PositionProvider,)
 
-    matching_functions = {
-        "urllib3.HTTPConnectionPool",
-        "urllib3.connectionpool.HTTPConnectionPool",
+    matching_functions: dict[str, str] = {
+        "urllib3.HTTPConnectionPool": "urllib3",
+        "urllib3.connectionpool.HTTPConnectionPool": "urllib3",
```

Comment:
I don't like this change. Or at least not how it's being used. It may be the case that we want to change it to a different function from another module (i.e. `a.call()` -> `b.func()`). 

I know it's still possible just point the function in the dict itself (e.g. `{a.call() : b.func()}`), as it is being used here, it makes the purpose of `matching_functions` parameter less clear.

In short, I'd prefer if you pointed the function itself here (i.e. `urllib3.HTTPConnectionPool : urllib3.HTTPSConnectionPool`) instead of the lib.

Reply:
I don't like this change either 😅 but I'm also not sure about the right way to encode this information. In fact, the https codemod doesn't use this at all so maybe there's a cleaner solution.